doc/policy/version3_transactions.md

```diff
@@ -0,0 +1,62 @@
+# Transactions with nVersion 3
+
+A transaction with its `nVersion` field set to 3 ("V3 transactions") are allowed in mempool and
+transaction relay.
+
+The policy rules set for V3 transactions is intended to aid contract or L2 protocols in which
+transactions are signed by untrusted counterparties well before broadcast time, e.g. the Lightning
+Network (LN). Since these contracting transactions are shared between multiple parties and mempool
+congestion is difficult to predict, mempool policy restrictions may accidentally allow a malicious
+party to pin a transaction.  The goal here is create a policy for V3 transactions that retains
+DoS-resistance and accepts transactions that are incentive-compatible to mine, while also avoiding
+specific pinning attacks.
+
+All existing standardness rules apply to V3. The following set of additional restrictions apply to
+V3 transactions:
+
+1. A V3 transaction can be replaced, even if it does not signal BIP125 replaceability. Other
+   conditions rules apply, see [RBF rules](./mempool-replacements.md) and [Package RBF
+rules][./packages.md#Package-Replace-By-Fee]. Use the (`-mempoolfullrbf`) configuration option to
+allow transaction replacement without enforcement of any opt-in signaling rule.
+
+2. Any descendant of an unconfirmed V3 transaction must also be V3.
+
+*Rationale*: Combined with Rule 1, this gives us the property of "inherited signaling" when
+descendants of unconfirmed transactions are created. Additionally, checking whether a transaction
+signals replaceability this way does not require mempool traversal, and does not change based on
+what transactions are mined.
+
+*Note*: The descendant of a *confirmed* V3 transaction does not need to be V3.
+
+3. A V3 transaction cannot have more than 1 descendant.
+
+*Rationale*: (upper bound) the larger the descendant limit, the more transactions may need to be
+replaced. This is a problematic pinning attack, i.e., a malicious counterparty prevents the
+transaction from being replaced by adding many descendant transactions that aren't fee-bumping.
+
+*Rationale*: (lower bound) at least 1 descendant is required to allow CPFP of the presigned
+transaction. The contract protocol can create presigned transactions paying 0 fees and 1 output for
+attaching a CPFP at broadcast time ("anchor output"). Without package RBF, multiple anchor outputs
+would be required to allow each counterparty to fee-bump any presigned transaction. With package
+RBF, since the presigned transactions can replace each other, 1 anchor output is sufficient.
+
+4. A V3 transaction that has an unconfirmed V3 ancestor cannot be larger than 1000 virtual bytes.
```

Comment:
Yeah good point, seems perfectly reasonable to me to have no payments in a CPFP transaction. I agree it seems like it would be fine to allow empty vout. But it looks like non-empty vout is indeed a consensus rule.
See `CheckTransaction`:
https://github.com/bitcoin/bitcoin/blob/291e363ce500e492475c4ccd189ea1d031c43613/src/consensus/tx_check.cpp#L16-L17
Which is called by `CheckBlock`:
https://github.com/bitcoin/bitcoin/blob/291e363ce500e492475c4ccd189ea1d031c43613/src/validation.cpp#L3370-L3381

Reply:
Extra trivia: seems like this consensus rule has been here since the very beginning :smiling_face_with_tear: can't believe Satoshi didn't predict the need for CPFPing presigned transactions. 

https://github.com/bitcoin/bitcoin/blob/4405b78d6059e536c36974088a8ed4d9f0f29898/main.h#L443-L448

Reply:
> Is there a consensus rule that disallows transactions with no outputs in blocks?

Yes. It's a hard rule.

Reply:
That's too bad! Damn you Satoshi, you could have thought about this...

Reply:
OP_RETURN outputs can be pretty small :)

Reply:
Mm, could have a 0-value OP_RETURN?

Reply:
yes, you'll pay 8+1+1 bytes for the output